head/src/main/kotlin/io/qalipsis/core/head/hook/PrePostLifecycleHook.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2022 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.core.head.hook
+
+import io.qalipsis.api.context.CampaignKey
+
+/**
+ * Hook to handle operations that should be triggered before campaign start and after campaign completion.
+ *
+ * @author Francisca Eze
+ */
+internal interface PrePostLifecycleHook {
+
+    /**
+     * Hook to set up certain processes before the start of a campaign.
+     */
+    suspend fun beforeStart(campaignKey: CampaignKey) = Unit
```

Comment:
Those hooks should be in the existing interface CampaignHook, there is no reason to multiply them.